sam.nim

```diff
@@ -71,7 +71,14 @@ proc findValue(m: Mapper, key: string, pos = 0): int {.noSideEffect.} =
 
 proc loads(target: var any, m: Mapper, pos = 0) =
   if pos < 0: return
-  when target.type is object or target.type is tuple:
+  when target.type is Option:
+    if m.tokens[pos].getValue(m.json) == "null":
```

Comment:
I don't know, maybe for performance it's better to check first that `kind == JSMN_PRIMITIVE`